tools/test/test_executorch_custom_ops.py

```diff
@@ -0,0 +1,74 @@
+import unittest
+
+from typing import Any, Dict
+
+from torchgen.executorch.api.custom_ops import ComputeNativeFunctionStub
+
+from torchgen.model import Location, NativeFunction
+
+SPACES = "    "
+
+
+def _get_native_function_from_yaml(yaml_obj: Dict[str, object]) -> NativeFunction:
+    native_function, _ = NativeFunction.from_yaml(
+        yaml_obj,
+        loc=Location(__file__, 1),
+        valid_tags=set(),
+    )
+    return native_function
+
+
+class TestComputeNativeFunctionStub(unittest.TestCase):
+    """
+    Could use torch.testing._internal.common_utils to reduce boilerplate.
+    GH CI job doesn't build torch before running tools unit tests, hence
+    manually adding these parametrized tests.
+    """
+
+    def _test_function_schema_generates_correct_kernel(
+        self, obj: Dict[str, Any], expected: str
+    ) -> None:
+        func = _get_native_function_from_yaml(obj)
+
+        gen = ComputeNativeFunctionStub()
+        res = gen(func)
+        self.assertEquals(
+            res,
+            expected,
+        )
+
+    def test_function_schema_generates_correct_kernel_tensor_out(self) -> None:
+        obj = {"func": "custom::foo.out(Tensor self, *, Tensor(a!) out) -> Tensor(a!)"}
+        expected = """
+at::Tensor & wrapper_out_foo_out(const at::Tensor & self, at::Tensor & out) {
+    return out;
+}
+    """
+        self._test_function_schema_generates_correct_kernel(obj, expected)
+
+    def test_function_schema_generates_correct_kernel_no_out(self) -> None:
+        obj = {"func": "custom::foo.Tensor(Tensor self) -> Tensor"}
+        expected = """
+at::Tensor wrapper_Tensor_foo(const at::Tensor & self) {
+    return self;
+}
```

Comment:
This tests are very awkward. At the very least, can we use an expect test for this?

Reply:
Any code pointer for existing expect tests?

Reply:
search assertExpectedInline